packages/blockchain-wallet-v4-frontend/src/components/QRCodeCapture/index.js

```diff
@@ -49,7 +49,7 @@ class QRCodeCaptureContainer extends React.PureComponent {
 
   handleScanBchAddress (data) {
     try {
-      if (utils.bch.isCashAddr(data)) {
+      if (utils.bitcoin.isValidBitcoinAddress(data)) {
```

Comment:
You are calling a BTC addr validation method when the outer function hah BCH in it's name. Is this intended?

Reply:
yes. bch should accept btc address.

Reply:
So then it should be, `utils.bitcoin.isValidBitcoinAddress(data) || utils.bch.isCashAddr(data)`? Or no.

Reply:
no because isCashAddr("bitcoincash:qpheqxjud2p0yg85t679tnc7n05xffs5ks07zhg6y6") evaluates to true, take a look at the screenshot in the issue.

so for bitcoin isValidBitcoinAddress returns true for plain addresses only and decode only works for prefixed address strings. for bitcoin cash both decode and isCash works for both, but decode works better because it correctly parses the address.

for bitcoin cash, we need to handle the case where the address is bitcoin address.

Reply:
So later on down in the function is where bch is handled correctly. Sorry for the confusion